### ISSUE TITLE
Unreleased: change in behaviour of converting Float to DateTime64

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#21034]
21.2:

```
milovidov-desktop :) SELECT CAST(1111111111.222 AS DateTime64(3))

SELECT cast(1111111111.222, 'DateTime64(3)')

Query id: d10c8623-5c74-4a38-b535-a99f6af4d28d

┌─cast(1111111111.222, 'DateTime64(3)')─┐
│               2005-03-18 04:58:31.222 │
└───────────────────────────────────────┘
```

master:

```
milovidov-desktop :) SELECT CAST(1111111111.222 AS DateTime64(3))

SELECT cast(1111111111.222, 'DateTime64(3)')

Query id: cb4af967-a4ae-4df5-89ba-234d68675f93

┌─cast(1111111111.222, 'DateTime64(3)')─┐
│               2005-03-18 04:58:31.000 │
└───────────────────────────────────────┘
```

**Comment by azat** [comment]
Yes, my bad (strange there were no tests), addressed here - #21050

**Comment by azat** [comment]
Cc: @Enmk
